build/tools/DockerCompose.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System.Collections.Generic;
+using Nuke.Common.IO;
+using Nuke.Common.Tooling;
+using Serilog;
+
+namespace Tools
+{
+    public static class DockerCompose
+    {
+        static string DockerComposePath =>
+            ToolPathResolver.GetPathExecutable("docker-compose");
+        
+        public static IReadOnlyCollection<Output> DockerComposeUp(AbsolutePath file, string arguments = "",  AbsolutePath workingDirectory = null)
+            => ExecuteDockerCompose(file, arguments, workingDirectory, "up");
+
+
+        public static IReadOnlyCollection<Output> DockerComposeDown(AbsolutePath file, string arguments = "", AbsolutePath workingDirectory = null)
+            => ExecuteDockerCompose(file, arguments, workingDirectory, "down");
+
+        public static IReadOnlyCollection<Output> DockerComposeBuild(AbsolutePath file, string arguments ="", AbsolutePath workingDirectory = null)
+            => ExecuteDockerCompose(file, arguments, workingDirectory, "build");
+
+        static IReadOnlyCollection<Output> ExecuteDockerCompose(AbsolutePath file, string userArguments, AbsolutePath workingDirectory, string command)
+        {
+            var arguments = string.Concat("-f ",file, " ", command, " ", userArguments);
+
+            var process = ProcessTasks.StartProcess(DockerComposePath, arguments, workingDirectory, customLogger: DockerLogger);
+            process.AssertZeroExitCode();
+            return process.Output;
+        }
+
+        static void DockerLogger(OutputType type, string output)
+        {
+            if (type == OutputType.Err && !output.Contains("failed"))
+            {
+                Log.Information(output);
```

Comment:
If `OutputType.Err`, perhaps `Log.Error/Warning` would be more appropriate? Also, why exclude output that contains `failed`?

Reply:
All output comes for DockerCompose is on StdError. So I have redirected it to the information stream of the logger unless it contains the word failed